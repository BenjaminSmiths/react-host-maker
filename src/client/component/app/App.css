.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.NavHeader {
  background-color: #444;
  height: 150px;
  padding: 20px;
  color: white;
}

.fixedHeader h2 {
  font-size: large;
  color: white;
}
.fix .NavHeader {
  padding: 0;
  height: 153px;
}
.fix .fixedHeader {
  background-color: #444;
  position: fixed;
  width: 101%;
  top: 0;
  padding: 0;
  height: 80px;
  z-index: 1;
  -webkit-box-shadow: 0 10px 14px -4px rgba(0, 0, 0, 0.43);
  -moz-box-shadow: 0 10px 14px -4px rgba(0, 0, 0, 0.43);
  box-shadow: 0 10px 14px -4px rgba(0, 0, 0, 0.43);
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
