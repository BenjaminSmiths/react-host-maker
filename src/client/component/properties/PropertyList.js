import React, {Component} from 'react';
import PropertyItem from './PropertyItem';
import {fetchProperties} from '../../store/properties/propertiesActions';
import {connect} from 'react-redux';
import './style.css';

class PropertyList extends Component {

    componentDidMount() {
        this.props.fetchProperties();
    }

    render() {
        return (
            <div name="PropertyList" className="propertyList container-fluid">
                <div className="container Title-header">
                    <h3>Your current listing's</h3>
                </div>
                {
                    this.props.properties && this.props.properties.map((p, i) => <PropertyItem property={p} key={i}/>)
                }
            </div>
        )
    }
}

export default connect(
    state => ({
        properties: state.properties
    }),
    {fetchProperties}
)(PropertyList);