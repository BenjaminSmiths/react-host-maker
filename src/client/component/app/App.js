import React from 'react';
import PropertyList from '../properties/PropertyList';
import Search from '../search/Search';
import NavBar from './NavBar';
import './App.css';

const App = () =>
    <div className="App">
        <NavBar />
        <PropertyList />
        <Search />
    </div>;

export default App;
