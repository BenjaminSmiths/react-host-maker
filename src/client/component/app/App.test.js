import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';
import PropertyList from '../properties/PropertyList';
jest.mock('react-redux', () => require('react-redux-mock'));
import { __setState } from 'react-redux';

describe('App', () => {

    beforeEach(() => {
        __setState({
            properties: [{
                'owner': 'carlos',
                'address': {
                    'line1': 'Flat 5',
                    'line4': '7 Westbourne Terrace',
                    'postCode': 'W2 3UL',
                    'city': 'London',
                    'country': 'U.K.'
                },
                'incomeGenerated': 2000.34,
                'image': 'property1.jpg'
            }]
        });
    });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
    });

    it('should say welcome', () => {
        let component = ReactTestUtils.renderIntoDocument(
            <App />
        );

        let h2 = ReactTestUtils.findRenderedDOMComponentWithTag(
            component, 'h2'
        );

        expect(ReactDOM.findDOMNode(h2).textContent).toEqual("Welcome to HostMaker");
    });

    it('should match the current snap', () => {
        const tree = renderer.create(
            <App />
        ).toJSON();
        expect(tree).toMatchSnapshot()
    });

    describe('list of properties', () => {

        it('should find a list component', () => {
            let component = ReactTestUtils.renderIntoDocument(
                <App />
            );

            let propertyList = ReactTestUtils.scryRenderedComponentsWithType(
                component, 'PropertyList'
            );

            expect(propertyList).toBeDefined();
        });
    })

});
