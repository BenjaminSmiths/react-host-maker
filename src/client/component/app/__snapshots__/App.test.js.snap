// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the current snap 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to HostMaker
    </h2>
  </div>
  <div
    className="propertyList container-fluid"
    name="PropertyList"
  >
    <div
      className="container Title-header"
    >
      <h3>
        Your current listing's
      </h3>
    </div>
    <div
      className="container propertyItem"
    >
      <div
        className="panel"
      >
        <div
          className="row headings"
        >
          <div
            className="col-xs-3"
          >
            Owner
          </div>
          <div
            className="col-xs-5"
          >
            Address
          </div>
          <div
            className="col-xs-3"
          >
            Income Generated
          </div>
        </div>
        <div
          className="row details"
        >
          <div
            className="owner col-xs-3"
          >
            Carlos
          </div>
          <div
            className="address col-xs-5"
          >
            <div>
              Flat 5
            </div>
            <div>
              7 Westbourne Terrace
            </div>
            <div>
              London, U.K.
            </div>
            <div>
              W2 3UL
            </div>
          </div>
          <div
            className="income col-xs-4"
          >
            Â£
            2, 000.34
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
